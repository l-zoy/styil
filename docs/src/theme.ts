import { createSystem, global, getCssValue } from '@styil/react'

const { styled } = createSystem({
  theme: (mode: 'light' | 'dark') => {
    return {
      dark: {
        bgColor: '#242424'
      },
      light: {
        bgColor: '#fff'
      }
    }[mode]
  },
  defaultMode: localStorage.getItem('styil-theme-mode') ?? 'light'
})

export { styled, global, getCssValue }
