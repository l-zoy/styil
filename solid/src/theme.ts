import { createSystem } from '../../dist/solid'

export const { styled, SystemProvider, useSystem, createExtracts } = createSystem({
  theme: (mode) => {
    return {
      light: {
        color: 'red'
      },
      dark: {
        color: 'blue'
      }
    }[mode]
  },
  defaultMode: 'light'
})
