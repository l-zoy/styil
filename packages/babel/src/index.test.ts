import plugin from '.'
import { transformSync } from '@babel/core'

describe('transform api', () => {
  it('sourcemap', () => {
    const code = `
      import {styled} from '@styils/react'
      const Button:string = styled('div', {
        color:'red'
      })

      render(<Button/>)
      `

    const res = transformSync(code, {
      plugins: [plugin],
      filename: 'test.js'
    })?.code

    expect(res).toMatch('sourceMappingURL=data:application/json;charset=utf-8;base64')
  })

  it('dynamic rendering', () => {
    const code = `
      import {styled} from '@styils/react'
      function test(){
        const Button = styled('div', {
          color:'red'
        })
      }
      `

    const res = transformSync(code, {
      plugins: [[plugin, { sourceFileName: 'test.js' }]]
    })?.code

    expect(res).toMatch('sourceMappingURL=data:application/json;charset=utf-8;base64')
    expect(res).toMatchSnapshot()
  })

  it('not sourceFileName', () => {
    const code = `
      import {styled} from '@styils/react'
      styled('div', {
        color:'red'
      })
      `

    try {
      transformSync(code, {
        plugins: [plugin]
      })?.code
    } catch (err: any) {
      expect(err.message).toEqual(
        'unknown: To compile using api, you need to pass in `sourceFileName`'
      )
    }
  })

  it('plugin already exists', () => {
    try {
      transformSync(``, {
        plugins: [plugin],
        parserOpts: {
          plugins: ['jsx', 'flow']
        }
      })
      expect(true).toEqual(true)
    } catch {
      expect(false).toEqual(true)
    }

    try {
      transformSync(``, {
        plugins: [plugin],
        parserOpts: {
          plugins: ['jsx', ['flow', {}]]
        }
      })
      expect(true).toEqual(true)
    } catch {
      expect(false).toEqual(true)
    }
  })

  it('precompiled', () => {
    const code = `
    import { styled1 } from '../theme';
    styled('div', {
      color: 'red'
    });`

    const res = transformSync(code, {
      plugins: [[plugin, { sourceFileName: 'test.js', importPaths: /theme$/ }]]
    })?.code

    expect(/styled\.sourceMap/.test(res!)).toEqual(false)

    const code1 = `
    import { styled } from '../theme';
    const Button = styled('div', {
      color: 'red'
    });`

    const res1 = transformSync(code1, {
      plugins: [[plugin, { sourceFileName: 'test.js', importPaths: /theme$/ }]]
    })?.code

    expect(/styled\.sourceMap/.test(res1!)).toEqual(true)

    const code2 = `
    import styled from '../theme';
    const Button = styled('div', {
      color: 'red'
    });`

    const res2 = transformSync(code2, {
      plugins: [[plugin, { sourceFileName: 'test.js', importPaths: /theme$/ }]]
    })?.code

    expect(res2).toMatch(`styled.sourceMap = "/*# sourceMappingURL=data:application/`)

    const code3 = `
    import styled from 'styled';
    styled('div', {
      color: 'red'
    });`

    const res3 = transformSync(code3, {
      plugins: [[plugin, { sourceFileName: 'test.js', importPaths: /theme$/ }]]
    })?.code

    expect(/styled\.sourceMap/.test(res3!)).toEqual(false)
  })

  it('global', () => {
    const code = `
    import {global} from '@styils/react'

    global({
      color:'red'
    })
    `

    const res = transformSync(code, {
      plugins: [[plugin, { sourceFileName: 'test.js' }]]
    })?.code

    expect(res).toMatchSnapshot()
  })

  it('global be wrapped', () => {
    const code = `
    import {global} from '@styils/react'

    if (true){
      global({
        color:'red'
      })
    }
    `

    const res = transformSync(code, {
      plugins: [[plugin, { sourceFileName: 'test.js' }]]
    })?.code

    expect(res).toMatchSnapshot()
  })
})
