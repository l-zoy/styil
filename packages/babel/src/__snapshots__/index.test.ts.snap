// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform api dynamic rendering 1`] = `
"import { styled } from '@styils/react';

function test() {
  process.env.NODE_ENV !== \\"production\\" && (styled.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBR3VCIiwiZmlsZSI6InRlc3QuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlc0NvbnRlbnQiOlsiXG4gICAgICBpbXBvcnQge3N0eWxlZH0gZnJvbSAnQHN0eWlscy9yZWFjdCdcbiAgICAgIGZ1bmN0aW9uIHRlc3QoKXtcbiAgICAgICAgY29uc3QgQnV0dG9uID0gc3R5bGVkKCdkaXYnLCB7XG4gICAgICAgICAgY29sb3I6J3JlZCdcbiAgICAgICAgfSlcbiAgICAgIH1cbiAgICAgICJdfQ== */\\")
  const Button = styled('div', {
    color: 'red'
  });
}"
`;

exports[`transform api global 1`] = `
"import { global } from '@styils/react';
process.env.NODE_ENV !== \\"production\\" && (global.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBR0kiLCJmaWxlIjoidGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzQ29udGVudCI6WyJcbiAgICBpbXBvcnQge2dsb2JhbH0gZnJvbSAnQHN0eWlscy9yZWFjdCdcblxuICAgIGdsb2JhbCh7XG4gICAgICBjb2xvcjoncmVkJ1xuICAgIH0pXG4gICAgIl19 */\\")
global({
  color: 'red'
});"
`;

exports[`transform api global be wrapped 1`] = `
"import { global } from '@styils/react';

if (true) {
  process.env.NODE_ENV !== \\"production\\" && (global.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInRlc3QuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBSU0iLCJmaWxlIjoidGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzQ29udGVudCI6WyJcbiAgICBpbXBvcnQge2dsb2JhbH0gZnJvbSAnQHN0eWlscy9yZWFjdCdcblxuICAgIGlmICh0cnVlKXtcbiAgICAgIGdsb2JhbCh7XG4gICAgICAgIGNvbG9yOidyZWQnXG4gICAgICB9KVxuICAgIH1cbiAgICAiXX0= */\\")
  global({
    color: 'red'
  });
}"
`;
